DOC: Simplify and update git setup page

bjnath opened the pull request:
Changed title from "Getting started with Git development"
to more specific "Setting up git for NumPy development"

Updated the outdated GitHub screenshot for forking button, added
more screenshots.

Simplified the narrative.

Fixed bad github links (always pointed to NumPy's github).

Clarified SSH setup, role, and effect on git remote -v.

Clarified meaning of "upstream" and "origin".

Replaced read of .git/config (platform-specific) with call to git config.

<!-- Please be sure you are following the instructions in the dev guidelines
http://www.numpy.org/devdocs/dev/development_workflow.html
-->

<!-- We'd appreciate it if your commit message is properly formatted
http://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message
-->

bjnath committed:
DOC: Simplify and update git setup page

Changed title from "Getting started with Git development"
to more specific "Setting up git for NumPy development"

Updated the outdated GitHub screenshot for forking button, added
more screenshots.

Simplified the narrative.

Fixed bad github links (always pointed to NumPy's github).

Clarified SSH setup, role, and effect on git remote -v.

Clarified meaning of "upstream" and "origin".

Replaced read of .git/config (platform-specific) with call to git config.

seberg reviewed (approved):
LGMT, I wonder if we should mention how to create and push a feature branch and keep the master branch up to date? In any case, I think its a big improvement, and I am happy to merge it pretty much as is.

bjnath commented on doc/source/dev/gitwash/development_setup.rst:
```diff
@@ -1,145 +1,172 @@
 
-.. _linking-to-upstream:
+#. Optionally, to get easy access to NumPy pull requests (useful mostly to
+   reviewers): ::
```
No need to keep it, then -- removed.

bjnath commented on doc/source/dev/gitwash/development_setup.rst:
```diff
@@ -1,145 +1,172 @@
-In detail
-=========
+   A directory ``numpy`` is created on your machine. (If you already have
+   a numpy directory, GitHub will choose a different name like ``numpy-1``.)
```
In this case I think it really is GitHub and not git. If you try to use git to do what GitHub is doing and clone a repo with a name you already have, git fails with `destination path xxx already exists and is not an empty directory.`

seberg replied:
Oh, interesting that the remote part has something to do with it

bjnath committed:
DOC: Add links, remove section from PR #16671

Per suggestion, removed outdated advice on getting pull branches.

Added back rST anchors, should have left them in.

bjnath commented:
>  I wonder if we should mention how to create and push a feature branch and keep the master branch up to date?

Yes -- this is development_workflow.rst, which I'll also take a look at. It references this page as a prerequisite.

> In any case, I think its a big improvement, and I am happy to merge it pretty much as is.

Thank you! Pushed a new version that removes the advice on pull branches. Left the GitHub -> git change (https://github.com/numpy/numpy/pull/16671#discussion_r445788167) alone -- see comment.

bjnath committed:
DOC: Add page anchor for PR #16671

Add '.. _development-setup:' page anchor.

bjnath commented:
The site has overlapping content on this subject (the first three bullets of https://numpy.org/devdocs/dev/gitwash/index.html and the start of https://numpy.org/devdocs/dev/index.html#development-process-summary). Noticed it just now. I'll submit another PR to clean that up.
